src/declarations/stencil-public-compiler.ts

```diff
@@ -424,7 +424,7 @@ type RequireFields<T, K extends keyof T> = T & { [P in K]-?: T[P] };
 /**
  * Fields in {@link Config} to make required for {@link ValidatedConfig}
  */
-type StrictConfigFields = 'flags' | 'logger' | 'outputTargets' | 'rootDir' | 'sys' | 'testing';
+type StrictConfigFields = 'flags' | 'logger' | 'outputTargets' | 'rootDir' | 'sys' | 'testing' | 'packageJsonFilePath';
```

Comment:
this adds `packageJsonFilePath` to the required fields, which requires changes every time we instantiate a `ValidatedConfig` and drops down a fair number of SNC errors just by itself.